src/index.js

```diff
@@ -646,17 +649,17 @@ function ncc (
 }
 
 // this could be rewritten with actual FS apis / globs, but this is simpler
-function getFlatFiles(mfsData, output, getAssetMeta, tsconfig, curBase = "") {
+function getFlatFiles(mfsData, output, getAssetMeta, compilerOptions, curBase = "") {
   for (const path of Object.keys(mfsData)) {
     const item = mfsData[path];
     let curPath = `${curBase}/${path}`;
     // directory
-    if (item[""] === true) getFlatFiles(item, output, getAssetMeta, tsconfig, curPath);
+    if (item[""] === true) getFlatFiles(item, output, getAssetMeta, compilerOptions, curPath);
     // file
     else if (!curPath.endsWith("/")) {
       const meta = getAssetMeta(curPath.slice(1)) || {};
-      if(curPath.endsWith(".d.ts")) {
-        const outDir = tsconfig.compilerOptions.outDir ? pathResolve(tsconfig.compilerOptions.outDir) : pathResolve('dist');
+      if (curPath.endsWith(".d.ts")) {
+        const outDir = compilerOptions?.compilerOptions?.outDir ? pathResolve(compilerOptions.compilerOptions.outDir) : pathResolve('dist');
```

Comment:
What is `compilerOptions?.compilerOptions` for?

Reply:
The latest commit resolves the problem. The review can be marked as resolved now.